frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx

```diff
@@ -250,8 +296,34 @@ export const OperatorHubSubscribeForm: React.FC<OperatorHubSubscribeFormProps> =
     };
 
     try {
-      if (isSuggestedNamespaceSelected && !suggestedNamespaceExists) {
-        await k8sCreate(NamespaceModel, ns);
+      if (isSuggestedNamespaceSelected) {
+        if (!suggestedNamespaceExists) {
+          await k8sCreate(NamespaceModel, ns);
+        } else {
+          const labels = {
+            ...suggestedNamespace.metadata.labels,
+            'openshift.io/cluster-monitoring': 'true',
+          };
+          const patch = [{ op: 'add', path: '/metadata/labels', value: labels }];
+          await k8sPatch(NamespaceModel, suggestedNamespace, patch);
+        }
+        if (
+          operatorRequestsMonitoring &&
+          (suggestedNamespaceRequestsMonitoring(suggestedNamespace) || enableMonitoring)
+        ) {
+          await k8sGet(RoleModel, rbacName, selectedTargetNamespace).catch((err) => {
```

Comment:
This assumes a certain naming convention, which might not be the case if the role was created outside console.

I'm not sure we should be trying to handle this for existing namespaces anyway.

Reply:
@spadgett so what should be the `Role` name that we want to create ? Can the 
`${selectedTargetNamespace}-prometheus;` be used ?

Reply:
Sounds fine to me. I was commenting more about trying to detect the role binding in an existing namespace.